src/Orleans.Core/Configuration/OptionLogger/GenericOptionFormatter.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.Extensions.Options;
+using Orleans.Runtime.Configuration;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Orleans
+{
+    internal class GenericOptionsFormatter<T> : IOptionFormatter<T>
+         where T : class, new()
+    {
+        public string Name { get; }
+
+        private T options;
+
+        public GenericOptionsFormatter(IOptions<T> options)
+        {
+            this.options = options.Value;
+            this.Name = OptionFormattingUtilities.Name<T>();
+        }
+
+        internal GenericOptionsFormatter(string name, T options)
+        {
+            this.options = options;
+            this.Name = OptionFormattingUtilities.Name<T>(name);
+        }
+
+        public IEnumerable<string> Format()
+        {
+            return typeof(T)
+                .GetProperties()
```

Comment:
I think we only want properties which are public, and have public getters and setters.  Something like

.GetProperties(BindingFlags.Public).where(prop => prop.GetGetMethod() != null && prop.GetSetMethod() != null)